packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart

```diff
@@ -24,14 +23,18 @@ String? _getUrlScheme(String url) => Uri.tryParse(url)?.scheme;
 bool _isSafariTargetTopScheme(String url) =>
     _safariTargetTopSchemes.contains(_getUrlScheme(url));
 
+bool _navigatorIsSafari(html.Navigator navigator) =>
+    navigator.userAgent.contains('Safari') &&
+    !navigator.userAgent.contains('Chrome');
```

Comment:
Now that we're not using `navigator.vendor` (or `appVersion`), please update the tests where we do things like this:

* https://github.com/flutter/packages/blob/c768b143db2875251ed38bf721a67b247644c392/packages/url_launcher/url_launcher_web/example/integration_test/url_launcher_web_test.dart#L151-L152

We should be able to only mock `userAgent`, and provide an user-agent for a recent Chrome and a recent Safari as well?